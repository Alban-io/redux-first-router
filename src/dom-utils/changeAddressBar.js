// @flow
import type { History, LocationState } from '../flow-types'


export default (
  locationState: LocationState,
  currentPathname: string,
  history: History,
): string => {
  if (locationState.pathname !== currentPathname) {
    history.push({ pathname: locationState.pathname })
  }

  return locationState.pathname
}
