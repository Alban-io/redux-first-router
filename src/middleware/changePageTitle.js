import changePageTitle from '../pure-utils/changePageTitle'
import isServer from '../pure-utils/isServer'

export default async (req, next) => {
  const title = req.getTitle()

  if (!isServer() && typeof title === 'string') {
    window.document.title = title
  }

  return next()
}
