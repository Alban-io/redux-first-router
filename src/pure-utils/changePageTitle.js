// @no-flow
// `CONFIG` global currently causes flow errors in user code
// see: http://stackoverflow.com/questions/42154415/in-flow-npm-packages-whats-the-proper-way-to-suppress-issues-so-user-apps-don

import type { Document } from '../flow-types'


export default (doc: Document, title: string): ?string => {
  if (typeof title === 'string' && doc.title !== title) {
    return doc.title = title
  }

  return null
}


export const getDocument = (): Document => {
  const isSSRTest = process.env.NODE_ENV === 'test'
    && typeof CONFIG !== 'undefined' && CONFIG.isSSR

  return typeof document !== 'undefined' && !isSSRTest
    ? document
    : {}
}
