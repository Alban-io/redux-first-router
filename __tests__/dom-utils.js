import changeAddressBar from '../src/dom-utils/changeAddressBar'
import changePageTitle from '../src/dom-utils/changePageTitle'


describe('changeAddressBar()', () => {
  it('when pathname changes push new pathname on to addressbar', () => {
    const locationState = { pathname: 'foo' }
    const currentPathname = 'bar'
    const history = []

    const ret = changeAddressBar(locationState, currentPathname, history)

    console.log(ret)
    console.log(history)

    expect(history).toEqual([{ pathname: 'foo' }])
    expect(ret).toEqual('foo')
  })

  it('when pathname does not change, do not push pathname on to address bar', () => {
    const locationState = { pathname: 'foo' }
    const currentPathname = 'foo'
    const history = []

    const ret = changeAddressBar(locationState, currentPathname, history)

    console.log(ret)
    console.log(history)

    expect(history).toEqual([])
    expect(ret).toEqual('foo')
  })
})


describe('changePageTitle()', () => {
  it('when title changes set it to document.title', () => {
    const document = { }
    const title = 'foo'

    const ret = changePageTitle(document, title)

    console.log(document)

    expect(document).toEqual({ title: 'foo' })
    expect(ret).toEqual('foo')
  })

  it('when title changes do not set document.title', () => {
    const document = { title: 'foo' }
    const title = 'foo'

    const ret = changePageTitle(document, title)

    console.log(document)

    expect(document).toEqual({ title: 'foo' })
    expect(ret).toEqual(null) // no return value when title does not change
  })
})
