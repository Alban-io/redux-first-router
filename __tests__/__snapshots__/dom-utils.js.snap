exports[`changeAddressBar sd66f 1`] = `1`;

exports[`changeAddressBar sd66ff 1`] = `2`;

exports[`changeAddressBar when pathname changes push new pathname on to addressbar 1`] = `2`;

exports[`changeAddressBar when pathname does not change, do not push pathname on to address bar 1`] = `5`;
